.github/actions/setup/action.yml

```diff
@@ -64,3 +64,7 @@ runs:
         if: ${{ inputs.java-version != '' }}
         shell: bash
         run: echo "::add-matcher::.github/problem-matcher.json"
+      - name: Free additional disk space
+        shell: bash
+        # Based on https://github.com/actions/runner-images/issues/2840
+        run: ./.github/bin/free-disk-space.sh
```

Comment:
How long does it take? Are we ok with adding this to all jobs, even when not needed? There are over 100 jobs.

Reply:
It's not adding much time. Seems to be around 2 minutes.

Reply:
That's 2 minutes per job, that would add to over 20 minutes for the whole run. Combined time for all jobs in a run is over 24 hours (see [this chart](https://nineinchnick.github.io/trino-cicd/reports/ci-cd/#runs-cumulative-job-duration-on-master-per-day)), so it's a small percentage, but since we're already at capacity, we can't afford this. We should make this step optional and only run it when necessary.

From https://github.com/trinodb/trino/actions/runs/6466992083/job/17556152500?pr=19327
```
Tue, 10 Oct 2023 08:55:50 GMT Run ./.github/bin/free-disk-space.sh
...
Tue, 10 Oct 2023 08:57:54 GMT Run actions/setup-java@v3
```

Reply:
How would you decide that we should run it or not?

Reply:
I assume you're working on this because we've noticed some jobs are failing because of this. So let's add it only to those jobs. So we can start with those jobs and enable this for others when we notice them failing. We already have lots of reports to track recurring job failures.